ESPHome/TX-Ultimate-Easy-ESPHome_core_hw_vibration.yaml

```diff
@@ -0,0 +1,120 @@
+#####                  Repository: https://github.com/edwardtfn/TX-Ultimate-Easy               #####
+####################################################################################################
+##### Purpose: ESPHome Core - Hardware - Vibration system                                      #####
+####################################################################################################
+##### Author: edwardtfn - https://github.com/edwardtfn - https://buymeacoffee.com/edwardfirmo  #####
+####################################################################################################
+##### NOTE:                                                                                    #####
+##### - Make changes ONLY if absolutely necessary and you have the required knowledge.         #####
+##### - For normal system use, modifications to this file are NOT required.                    #####
+####################################################################################################
+---
+binary_sensor:
+  - id: bs_vibrating
+    name: Vibrating
+    icon: mdi:vibrate
+    internal: false
+    platform: template
+    lambda: return sw_vibration_motor->state;
+
+button:
+  - id: bt_vibrate
+    name: Vibrate
+    icon: mdi:vibrate
+    internal: false
+    platform: template
+    on_press:
+      then:
+        - script.execute: vibrate
+
+number:
+  - id: nr_vibrating_duration
+    name: Vibrate duration
+    icon: mdi:vibrate
+    unit_of_measurement: ms
+    # mode: box
+    internal: false
+    entity_category: config
+    platform: template
+    min_value: 1
+    max_value: 500
+    step: 1
+    initial_value: 15
+    optimistic: true
+    restore_value: true
+
+script:
+  - id: !extend touch_on_multi_touch_release
+    then:
+      - script.execute: touch_on_multi_touch_release_vibration
+
+  - id: touch_on_multi_touch_release_vibration
+    mode: restart
+    then:
+      - lambda: |-
+          if (sl_touch_vibration_feedback->state != "Disabled")
+            vibrate->execute();
+
+  - id: !extend touch_on_press
+    then:
+      - script.execute: touch_on_press_vibration
+
+  - id: touch_on_press_vibration
+    mode: restart
+    then:
+      - lambda: |-
+          if (sl_touch_vibration_feedback->state == "On press" or sl_touch_vibration_feedback->state == "Always")
+            vibrate->execute();
+
+  - id: !extend touch_on_release
+    then:
+      - script.execute: touch_on_release_vibration
+
+  - id: touch_on_release_vibration
+    mode: restart
+    then:
+      - lambda: |-
+          if (sl_touch_vibration_feedback->state == "On release" or sl_touch_vibration_feedback->state == "Always")
+            vibrate->execute();
+
+  - id: vibrate
+    mode: restart
+    then:
+      - switch.turn_on: sw_vibration_motor
+      - delay: !lambda return nr_vibrating_duration->state;
+      - switch.turn_off: sw_vibration_motor
+
```

Comment:
No protection needed here... If the method is called again, it will restart the counter and then turn-off anyways. In case of a restart while the vibration is on, it will be turned off by `restore_mode: ALWAYS_OFF`.
In addition, I'm adding a turn-off when the OTA starts:
```yaml
ota:
  on_state_change:
    then:
      - switch.turn_off: sw_vibration_motor
```